library/packages/src/modules/Package.rb

```diff
@@ -221,6 +222,22 @@ def reset
       @removed_packages.clear
     end
 
+    # Tries to find a package according to the pattern
+    #
+    # @param pattern [String] a regex pattern to match, no escaping done
+    # @return list of matching package names
+    def by_pattern(pattern)
+      raise ArgumentError, "Missing search pattern" if pattern.nil? || pattern.empty?
+
+      # NOTE: - Resolvable.find takes POSIX regexp, later select uses Ruby regexp
+      # - Resolvable.find supports regexps only for dependencies, so we need to
+      # filter result according to package name
+      Y2Packager::Resolvable.find({ provides_regexp: "^#{pattern}$" }, [:name])
```

Comment:
Two problems here:

- It does not specify the kind of the resolvable, this might return any matching objects including patterns or products (i.e. not only packages).
- The `:name` attribute is always loaded, you do not need to specify that (minor issue) 

```suggestion
      Y2Packager::Resolvable.find({ kind: :package, provides_regexp: "^#{pattern}$" })
```


However, the function seems to be very specific, it searches Provides **and** the name using the same regexp. This looks like a very special case which very likely won't be used by anyone else. I'm not sure if it makes sense to share it here, I'd move it to the code which needs this.

Finally, the method name looks a bit confusing, "pattern" could be easily confused with libzypp patterns. A better name would be `by_provides_regexp` or something like that...